gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -278,18 +318,27 @@ public void copy(Path source, Path target, CopyOption... options) throws IOExcep
       } else if (option instanceof CloudStorageOption) {
         if (option instanceof OptionBlockSize) {
           blockSize = ((OptionBlockSize) option).size();
+        } else if (option instanceof OptionMimeType) {
+          tgtInfoBuilder.contentType(((OptionMimeType) option).mimeType());
+          setContentType = true;
+        } else if (option instanceof OptionCacheControl) {
+          tgtInfoBuilder.cacheControl(((OptionCacheControl) option).cacheControl());
+          setCacheControl = true;
+        } else {
```

Comment:
What about `OptionUserMetadata`, `OptionContentEncoding`, and `OptionContentDisposition`?